idea/test-balance/src/common/transfer-balance-process.ts

```diff
@@ -4,21 +4,48 @@ import { transferService } from '../services/transfer.service';
 import { gearService } from '../gear';
 import { testBalanceLogger } from './test-balace.logger';
 import { producer } from '../rabbitmq/producer';
+import { EventEmitter } from 'node:events';
 
-export async function transferBalanceProcess(payload: any, correlationId: string): Promise<void> {
-  let result;
+export const requests: Array<{ payload: any; correlationId: string }> = [];
+const pushEmitter = new EventEmitter();
 
-  try {
-    const isPossibleToTransfer = await transferService.isPossibleToTransfer(payload.address, payload.genesis);
-    if (isPossibleToTransfer) {
-      const transferBalance = await gearService.transferBalance(payload.address);
-      result = { result: transferBalance };
-    } else {
-      result = { error: JSONRPC_ERRORS.TransferLimitReached.name };
+Object.defineProperty(requests, 'push', {
+  value: function (v: any) {
+    this[this.length] = v;
+    pushEmitter.emit('push');
+  },
+});
+
+async function* transferGenerator() {
+  while (true) {
+    if (requests.length === 0) {
+      await new Promise((resolve) => {
+        pushEmitter.once('push', resolve);
+      });
+    }
+    const req = requests.shift();
+    if (req) {
+      yield req;
+    }
+  }
+}
+
+export async function transferProcess(): Promise<void> {
+  for await (const { payload, correlationId } of transferGenerator()) {
+    let result;
+
```

Comment:
Small suggestion

```javascript
const { address, genesis } = payload;

const isPossibleToTransfer = await transferService.isPossibleToTransfer(address, genesis);

const transferBalance = await gearService.transferBalance(address);

```

Reply:
doesn't matter

Reply:
It matters for future people who will read your code easily not only for you